src-self-hosted/link/MachO.zig

```diff
@@ -225,66 +227,62 @@ pub fn flush(self: *MachO, module: *Module) !void {
 
     switch (self.base.options.output_mode) {
         .Exe => {
-            if (self.link_against_system) {
-                if (is_darwin) {
-                    {
-                        // Specify path to dynamic linker dyld
-                        const cmdsize = commandSize(@intCast(u32, @sizeOf(macho.dylinker_command) + mem.lenZ(DEFAULT_DYLD_PATH)));
-                        const load_dylinker = [1]macho.dylinker_command{
-                            .{
-                                .cmd = macho.LC_LOAD_DYLINKER,
-                                .cmdsize = cmdsize,
-                                .name = @sizeOf(macho.dylinker_command),
-                            },
-                        };
-                        try self.commands.append(self.base.allocator, .{
+            if (is_darwin) {
```

Comment:
@andrewrk OK, so I've decided to take the "adjust target version range" route for Darwin as I can't find any link between OS version and `libSystem` version. In effect, I've added `DarwinVersionRange` struct similar to `LinuxVersionRange` which in addition to storing the OS version, it now also stores the `libSystem` version as well. By default, the lib's version is currently set at `1281.100.1`, and if we're running on macOS natively, this version will get auto-detected in `NativeTargetInfo`, `.macosx` prong using `std.c.NSVersionOfRunTimeLibrary`. Since this info is now part of `Target` struct, Mach-O implementation doesn't have to call in to the OS anymore and doesn't require any OS-specific prongs.

Let me know what you think about this proposal!

Reply:
I'd like to explore the option of determining the libSystem version based on the status quo Darwin version range a little bit more. Here's my current mental model of the situation:

 * Our current minimum supported macOS version is 10.13 (high sierra). Soon (in fact overdue) to be bumped to 10.14 (mojave). The only other versions in existence are 10.15 (catalina) and 11.0 (big sur).
 * When each of these OS's were released, if you query the libSystem version, you would get some value. That value is the mapping from Darwin version to libSystem version.
 * Using an older libSystem version is harmless. We can hard code the libSystem version from our minimum supported macOS version, and it will work fine for all subsequent OS versions. This is preferable to modifying the status quo target OS version range scheme.

Reply:
I’m generally happy with this, however, I’d like to move forward with this PR which is blocking me on the linker implementation. Here’s my proposition. Since working out the correct value/pattern for libSystem might take some time (unless you have a specific idea for this), would it be OK to revert the changes in the latest commit and hardcode a lib version in the linker for now and merge this PR while we figure out the pattern? This would definitely help me out in my linker work.

Reply:
Oh, in fact, maybe an issue would be good for this? Perhaps we’d get others opinions in?

Reply:
And maybe I should have led with this: do you know of some resource which would list all versions of libSystem? Also, do you remember/know what the libSystem version for 10.13 was?

Reply:
Hard coding in a lib version for now sounds like the right approach. The only thing I'm hard vetoing here is doing native system detection (`is_darwin`) in the linker code. If you want to change the target version range schematic, let's go through the proposal process for that since it's a pretty significant breaking change.

Does that all make sense?

Reply:
As for obtaining that list, I have catalina. I bet if we make a small zig snippet and ask community members we could get people to run it on their machines and report the number.

Reply:
> Hard coding in a lib version for now sounds like the right approach. The only thing I'm vetoing here is doing native system detection (`is_darwin`) in the linker code. If you want to change the target version range schematic, let's go through the proposal process for that since it's a pretty significant breaking change.
> 
> Does that all make sense?

Absolutely, and I agree 100%. Latest changes in forced-push commit 55dc81b.

Reply:
> As for obtaining that list, I bet if we make a small zig snippet and ask community members we could get people to run it on their machines and report the number. I have catalina, maybe others have the other versions.

That's a brilliant idea!